python/looker/rtl/user_session.py

```diff
@@ -1,250 +1,108 @@
-        return self.active_token()
-
-    def login(self, settings=None) -> looker.models.access_token:
-        """Authenticate and login user to the API with default or specified settings"""
-        if settings is not None:
-            self.settings = settings
-
-        self._login()
-
-        if self.settings.user_id:
-            self.login_user(self.settings.user_id)
-
-        return self.active_token()
-
-    def logout(self):
-        """Logs out the active user. If impersonating, logs out impersonated user and logs
-        in with client_id and client_secret
-        """
-        if self.is_authenticated():
-            self.auth_api.logout()
-            if self.impersonating():
-                # clear impersonation settings
-                self._user_id = None
-                self._user_token = None
-                # log back in using client id and secret only
-                self._login()
-            else:
-                self.reset()
-
-    def login_user(self, user_id) -> looker.models.access_token:
-        self.user_id = user_id
-        self._user_token = self.auth_api.login_user(user_id)
-        if self._user_token is not None:
-            self.assign_api_client(self.user_token)
-            self.set_auth_expiration(self.user_token)
-        return self.active_token()
-
-    def logout_user(self):
-        if self.user_id is None:
-            raise SDKError("No user impersonation is active")
-        return self.api_client.logout()
-
-
-if __name__ == "__main__":
-    session = UserSession()
-    token: looker.models.access_token = session.login()
-    print(session.me)
-    print(token)
-    session.logout()
+    def __login(self, user_id: Optional[str]) -> tp.SDKSuccessResponse.value:
+        if (user_id and self.user_id != user_id):
+            # We are switching user ids
+            self.__logout()
+
+        self.__reset()
+
+        # pylint: disable=too-many-function-args
+        result = self.__ok(self.transport.request(tp.HttpMethod.POST,
+                                                  '/login',
+                                                  body={
+                                                      'client_id': self.settings.client_id,
+                                                      'client_secret': self.settings.client_secret
+                                                  }))
+        return result
+
+    def logout(self) -> bool:
+        result = False
+        if self.is_authenticated:
+            result = self.__logout()
+        return result
+
+    def __logout(self):
+        token = self._token
+        # TODO: this is still to be built based on authenticator
+
+    def __reset(self):
+        self._token = at.AuthToken()
+
+    def __ok(self, response: Union[tp.SDKErrorResponse, tp.SDKSuccessResponse]
+            ) -> Union[tp.SDKErrorResponse, tp.SDKSuccessResponse.value]:
+        if response.ok:
+            return response.value
+        raise tp.SDKErrorResponse
```

Comment:
hm, not sure this will work since `SDKErrorResponse` isn't an error class. looks like in typescript land @jkaster created an `sdkError` funciont that returns an error instance. I say you add an `SDKError` class to the `transport` module and raise it here with the `response.value` as the message